security/azure/restrict_rdp_internet/azure_restrict_rdp_inet.pt

```diff
@@ -0,0 +1,203 @@
+      field "securityRules", jmes_path(col_item, "properties.securityRules")
+      field "provisioningState", jmes_path(col_item,"properties.provisioningState")
+      field "description", jmes_path(col_item,"properties.description")
+      field "protocol", jmes_path(col_item,"properties.protocol")
+      field "destinationPortRange", jmes_path(col_item,"properties.destinationPortRange")
+      field "access", jmes_path(col_item,"properties.access")
+      field "direction", jmes_path(col_item,"propteries.direction")
+      field "sourceAddressPrefix", jmes_path(col_item,"properties.sourceAddressPrefix")
+    end
+  end
+end
+
+datasource "ds_filter_azure_resources" do
+  run_script $js_filter_azure_resources, $ds_azure_resources
+end
+
+
+###############################################################################
+# Scripts
+###############################################################################
+
+script "js_filter_azure_resources", type: "javascript" do
+  result "res"
+  parameters "ds_azure_resources"
+  code <<-EOS
+  var res = [];
+  _.each(ds_azure_resources, function (ds_azure_resource) {
+    bad_resource = false;
+    _.each(ds_azure_resource.securityRules, function (security_rule) {
+      if (security_rule.properties.access === "Allow") {
+        if (security_rule.properties.direction === 'Inbound') {
+          if (security_rule.properties.protocol === 'TCP') {
+            var bad_port = false;
+            var destinationPortRange = security_rule.properties.destinationPortRange.split('-');
+            if (destinationPortRange[1] != undefined) {
+              if (destinationPortRange[0] <= 3389 && destinationPortRange[1] >= 3389) {
+                bad_port = true;
+                console.log(destinationPortRange[0]);
+              }
+            } else {
+              if (destinationPortRange[0] == '3389' || destinationPortRange[0] == '*') {
+                bad_port = true;
+              }
+            }
+            if (bad_port == true) {
+              if (_.contains(["*", "0.0.0.0", "<nw>/0", "/0", "internet", "Any"], security_rule.properties.sourceAddressPrefix)) {
+                bad_resource = true;
+              }
+            }
+          }
+        }
+      }
+    });
+    if (bad_resource) {
+      res.push(ds_azure_resource);
+    }
+
+  });
+  EOS
+end
+
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_azure_resource" do
+  validate $ds_filter_azure_resources do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} rows containing Azure network security groups with RDP exposed to the internet."
+    escalate $escalate_email
+    check eq(size(data),0)
+    export do
+      resource_level true
+      field "id" do
+        label "Azure Resource Id"
+      end
+      field "name" do
+        label "Resource Name"
```

Comment:
Change to something more specific/descriptive, such as "Network Security Group Name"